src/toil_scripts/gatk_processing/gatk_preprocessing.py

```diff
@@ -7,433 +7,239 @@
 
 1 = download shared data
 2 = reference preprocessing
-3 = download sample 
-4 = index
-5 = sort
-6 = mark duplicates
-7 = index
-8 = realigner target 
-9 = indel realignment
-10 = index
-11 = base recalibration
-12 = ouput bqsr file
+3 = download sample
+4 = remove secondary alignments
+5 = index sample
+6 = sort sample
+7 = mark duplicates
+8 = index
+9 = realigner target
+10 = indel realignment
+11 = index
+12 = base recalibration
+13 = output bqsr fil6
 """
 import argparse
-import base64
-import collections
-import errno
-import hashlib
-import logging
 import multiprocessing
 import os
-import shutil
-import subprocess
 import sys
+import logging
 import textwrap
-
 import yaml
+
+from copy import deepcopy
+from urlparse import urlparse
+
 from toil.job import Job
-from toil_scripts import download_from_s3_url
 from toil_scripts.lib import require
+from toil_scripts.lib.files import copy_to
+from toil_scripts.lib.urls import download_url_job, s3am_upload
```

Comment:
`toil_scripts.lib.urls` after `toil_scripts.lib.programs`, `s3am_upload` is imported twice in this file.